src/snakemake/template_rendering/__init__.py

```
@@ -65,7 +65,7 @@ def check_template_output(job):
                 if f.is_storage and f in l:
                     raise WorkflowError(
                         "Output of template_engine rule contains local path to input file "
-                        f"from storage: {f} for {f.storage_object.query}. "
+                        f"from storage: {fmt_iofile(f)} for {f.storage_object.query}. "
```

Ruff (0.8.2)

Undefined name `fmt_iofile`

(F821)